pallets/utxo/src/lib.rs

```diff
@@ -563,6 +679,58 @@ pub mod pallet {
         Ok(().into())
     }
 
+    pub fn token_create<T: Config>(
+        caller: &T::AccountId,
+        public: H256,
+        input_for_fee: TransactionInput,
+        token_name: String,
+        token_ticker: String,
+        supply: Value,
+    ) -> Result<u64, DispatchErrorWithPostInfo<PostDispatchInfo>> {
+        ensure!(token_name.len() <= 25, Error::<T>::Unapproved);
+        ensure!(token_ticker.len() <= 5, Error::<T>::Unapproved);
+        ensure!(!supply.is_zero(), Error::<T>::MinBalanceZero);
+
+        // Take a free TokenID
+        let token_id =
+            <TokensHigherID<T>>::get().checked_add(1).ok_or("All tokens IDs has taken")?;
+
+        // Input with MLT FEE
+        let fee = UtxoStore::<T>::get(input_for_fee.outpoint).ok_or(Error::<T>::Unapproved)?.value;
+        ensure!(fee >= 100, Error::<T>::Unapproved);
```

Comment:
This still needs changing. The fee should be >= 100 * 1_000 * 100_000_000

Reply:
Okay let me explain

Reply:
This deserves a constant. Or, to be even more defensive, a special type (idk maybe `MTLAmount`) with two constructors like `from_mlt` and `from_mintseed` so we are explicit about the unit used everywhere.

Reply:
1 MLT = 1 * 1_000 * 100_000_000 base units, let's call them pennies.
Because we need MLT to be divisible and we don't handle floating point numbers the base unit, the penny, is the smallest division of the MLT. 

If a user sends 1 MLT they actually send 1 * 1_000 * 100_000_000 base units/pennies. To send 500 MLT they'd send 500 * 1_000 * 100_000_000 base units/pennies

The 100 unit fee here is actually 100/(1_000 * 100_000_000) which is what a nano-MLT...

Think of bitcoin, there's no such thing as a bitcoin in the code base, everything is in sats and the idea of a bitcoin is essentially a UI idea. https://www.investopedia.com/terms/s/satoshi.asp

Reply:
https://github.com/mintlayer/core/blob/36599cea0eafbac785a89ced09f0fdb3233b1042/node/src/chain_spec.rs#L150

Check here where we create 400_000_000 coins in the genesis block

Reply:
Fixed